examples/appending_events.rs

```diff
@@ -0,0 +1,154 @@
+    };
+
+    let event = EventData::json("some-event", &data)?.id(Uuid::new_v4());
+    let options = AppendToStreamOptions::default();
+
+    let _ = client
+        .append_to_stream("same-event-stream", &options, event.clone())
+        .await?;
+
+    let _ = client
+        .append_to_stream("same-event-stream", &options, event)
+        .await?;
+
+    // endregion append-duplicate-event
+
+    Ok(())
+}
+
+pub async fn append_with_no_stream(client: &Client) -> Result<()> {
+    // region append-with-no-stream
+
+    let data = TestEvent {
+        id: "1".to_string(),
+        important_data: "some value".to_string(),
+    };
+
+    let event = EventData::json("some-event", &data)?.id(Uuid::new_v4());
+    let options = AppendToStreamOptions::default().expected_version(ExpectedVersion::NoStream);
+
+    let _ = client
+        .append_to_stream("same-event-stream", &options, event)
+        .await?;
+
+    let data = TestEvent {
+        id: "2".to_string(),
+        important_data: "some other value".to_string(),
+    };
+
+    let event = EventData::json("some-event", &data)?.id(Uuid::new_v4());
+
+    let _ = client
+        .append_to_stream("same-event-stream", &options, event)
+        .await?;
+
+    // endregion append-with-no-stream
+
+    Ok(())
+}
+
+pub async fn append_with_concurrency_check(client: Client) -> Result<()> {
+    // region append-with-concurrency-check
+
+    let options = ReadStreamOptions::default().origin(Origin::End);
+
+    let last_event = client
+        .read_stream("concurrency-stream", &options, Single)
+        .await?
+        .ok()
+        .expect("we expect the stream to at least exist.")
+        .expect("we expect the stream to have at least one event.");
+
+    let data = TestEvent {
+        id: "1".to_string(),
+        important_data: "clientOne".to_string(),
+    };
+
+    let event = EventData::json("some-event", data)?.id(Uuid::new_v4());
+    let options = AppendToStreamOptions::default().expected_version(ExpectedVersion::Exact(
+        last_event.get_original_event().revision,
+    ));
+
+    let _ = client
+        .append_to_stream("concurrency-stream", &options, event)
+        .await?;
+
+    let data = TestEvent {
+        id: "2".to_string(),
+        important_data: "clientTwo".to_string(),
+    };
+
+    let event = EventData::json("some-event", &data)?.id(Uuid::new_v4());
+
+    let _ = client
+        .append_to_stream("concurrency-stream", &options, event)
+        .await?;
+
+    // endregion append-with-concurrency-check
+
+    Ok(())
+}
+
+pub async fn append_overriding_user_credentials(client: &Client) -> Result<()> {
+    let data = TestEvent {
+        id: "1".to_string(),
+        important_data: "clientOne".to_string(),
+    };
+
+    let event = EventData::json("some-event", data)?.id(Uuid::new_v4());
+    let options =
+        AppendToStreamOptions::default().authenticated(Credentials::new("admin", "changeit"));
+
+    // region overriding-user-credentials
```

Comment:
**FIX:** Could you move options creation to be within this region? It's the most important piece to show how to construct them (other samples are also showing that).

Reply:
Done